pkg/authorization/authorizer/scope/converter.go

```diff
@@ -384,45 +419,46 @@ func (e clusterRoleEvaluator) ResolveGettableNamespaces(scope string, clusterPol
 		ResourceRequest: true,
 	}
 
-	errors := []error{}
-	for _, rule := range rules {
-		matches, err := authorizer.RuleMatches(attributes, rule)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
-		if matches {
-			return []string{scopeNamespace}, nil
-		}
+	if authorizerrbac.RulesAllow(attributes, rules...) {
+		return []string{scopeNamespace}, nil
 	}
 
-	return []string{}, kutilerrors.NewAggregate(errors)
+	return []string{}, nil
 }
 
 // TODO: direct deep copy needing a cloner is something that should be fixed upstream
 var localCloner = conversion.NewCloner()
 
+func remove(array []string, foreign string) []string {
+	newar := array[:0]
+	for _, element := range array {
+		if element != foreign {
+			newar = append(newar, element)
```

Comment:
This is tricky to read.   I would expect it to allocate new instead of appending in place since I *think* the slice is detected to have a zero capacity.  Let @liggitt fiddle with it to make himself happy (you can make him an issue).  Mind giving me the new allocation version of this method (`newar = make([]string, ....`) for merge?